node/network/availability-distribution/src/lib.rs

```diff
@@ -296,8 +296,9 @@ impl ProtocolState {
 				// remove from the ancestry index
 				self.ancestry.remove(relay_parent);
 				// and also remove the actual receipt
-				self.receipts.remove(relay_parent);
-				self.per_candidate.remove(relay_parent);
+				if let Some(candidates) = self.receipts.remove(relay_parent) {
+					candidates.into_iter().for_each(|c| { self.per_candidate.remove(&c.0); });
```

Comment:
I think this is actually a bug, as a pending candidate could exists for multiple relay parents? Maybe I'm wrong, I didn't looked closer for now. I just tried to replicate the old behavior.

Reply:
`CandidateReceipt` includes the `CandidateDescriptor` field that in turn references a `relay_parent`. So two candidates with multiple relay parents will not hash to the same value

Reply:
`if descendants.is_empty() {` should prevent this, but given your changes have to look into it again.